### DISCUSSION TITLE
Gleam syntax highlight broken after guard with !

### DISCUSSION THREAD

**Discussion by darky** [github-linguist/linguist#6953] [Classification]
Example

https://github.com/darky/glerd/blob/6b7f4a9e143c82bc5a24bcf2256f92c0e154a103/src/glerd.gleam#L91-L206

**Comment by lildude** [comment]
From the [docs](https://github.com/github-linguist/linguist/blob/master/docs/troubleshooting.md#theres-a-problem-with-the-syntax-highlighting-of-a-file):

> ## There's a problem with the syntax highlighting of a file
>
> Linguist detects the language of a file but the actual syntax-highlighting is powered by a set of language grammars which are included in this project as a set of submodules [as listed here](/vendor/README.md).
>
> If you experience an issue with the syntax-highlighting on GitHub, **please report the issue to the upstream grammar repository, not here.**
Grammars are updated every time we build the Linguist gem so upstream bug fixes are automatically incorporated as they are fixed.

If this is reproducible, then it needs to be reported to the grammar maintainers. If it's just that page, then it's likely a rendering issue outside of linguist's control and needs to be reported to GitHub's support using the Contact link at the bottom of the page.

**Reply by darky** [reply]
Gleam creator says, that Github uses a outdated version of grammar

https://github.com/gleam-lang/tree-sitter-gleam/issues/93#issuecomment-2242659259
